lib/Serverless.test.js

```diff
@@ -115,6 +115,14 @@ describe('Serverless', () => {
       expect(serverless.cli).to.be.instanceof(CLI);
     });
 
+    it('should allow a custom CLI instance', () => {
+      class CustomCLI extends CLI {}
+      serverless.classes.CLI = CustomCLI;
+      serverless.init();
```

Comment:
Here we should return the Promise. Otherwise this test might be non-deterministic.

Reply:
Nice! Just getting started with serverless, previously had ansible playbook and then a messy aws-sdk setup so looking forward to a serverless future.

Reply:
Great 👍! Thanks for the feedback. Great to hear that you give it a spin!

And thanks for the PR @patrickheeney 💯